meteor/client/ui/DndListItemWrapper.tsx

```diff
@@ -0,0 +1,87 @@
+import { DragDropItemTypes } from './DragDropItemTypes'
+import React, { useRef } from 'react'
+import { DragSourceMonitor, DropTargetMonitor, useDrag, useDrop, XYCoord } from 'react-dnd'
+import { Identifier } from 'dnd-core'
+import { faGripLines } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+
+interface IDndItemWrapperProps {
+	index: number
+	dndType: DragDropItemTypes
+	listItem: any
+	className: string
+
+	onDropOutside: () => void
+	moveHandler: (dragIndex: number, hoverIndex: number) => void
+}
+
+interface DraggableItem {
+	index: number
+	type: DragDropItemTypes
+}
+
+export const DndListItemWrapper: React.FunctionComponent<IDndItemWrapperProps> = (props: IDndItemWrapperProps) => {
+	const ref = useRef<HTMLTableRowElement>(null)
+	const index = props.index
+
+	const [{ handlerId }, drop] = useDrop<DraggableItem, void, { handlerId: Identifier | null }>({
+		accept: props.dndType,
+		collect: (monitor: DropTargetMonitor) => ({ handlerId: monitor.getHandlerId() }),
+		hover(variant: DraggableItem, monitor: DropTargetMonitor) {
+			if (!ref.current) {
+				return
+			}
+			const dragIndex = variant.index
+			const hoverIndex = index
+
+			if (dragIndex === hoverIndex) {
+				return
+			}
+
+			const hoverBoundingRect = ref.current?.getBoundingClientRect()
+			const hoverMiddleY = (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2
+			const clientOffset = monitor.getClientOffset()
+			const hoverClientY = (clientOffset as XYCoord).y - hoverBoundingRect.top
+
+			if (dragIndex < hoverIndex && hoverClientY < hoverMiddleY) {
```

Comment:
Extracting these conditionals into aply named methods would improve the readability a lot:
```
if (isOutOfBounds) {
  return
}

isOutOfBounds(...): boolean {
  return (dragIndex < hoverIndex && hoverClientY < hoverMiddleY) 
    || (dragIndex > hoverIndex && hoverClientY > hoverMiddleY)
}
```

Maybe still have them as two separate conditionals. Example only has one because I can't see what they actually are checking (another reason for why a good name would be nice)

Reply:
After examining closer i believe that some of the functionality was unnecessary and already handled, the method has been greatly simplified.